src/localize.cpp

```diff
@@ -264,62 +262,38 @@ void localize_wrk(const BasisSet & basis, arma::mat & C, arma::vec & E, const ar
       fname="";
   }
 
-  arma::cx_mat Cplx(C*std::complex<double>(1.0,0.0));
-
-  // Save indices
-  std::vector<size_t> printidx(locorb);
-
-  // Loop over orbitals
-  while(locorb.size()) {
-    // Orbitals to treat in this run
-    std::vector<size_t> orbidx;
-
-    // Occupation number
-    double occno=occs[locorb[0]];
-
-    for(size_t io=locorb.size()-1;io<locorb.size();io--)
-      // Degeneracy in occupation?
-      if(fabs(occs[locorb[io]]-occno)<1e-6) {
-	// Orbitals are degenerate; add to current batch
-	orbidx.push_back(locorb[io]);
-	locorb.erase(locorb.begin()+io);
-      }
-
-    std::sort(orbidx.begin(),orbidx.end());
-
-    // Collect orbitals
-    arma::mat Cwrk(C.n_rows,orbidx.size());
-    for(size_t io=0;io<orbidx.size();io++)
-      Cwrk.col(io)=C.col(orbidx[io]);
-    // and orbital energies
-    arma::vec Ewrk(orbidx.size());
-    for(size_t io=0;io<orbidx.size();io++)
-      Ewrk(io)=E(orbidx[io]);
-
-    // Localizing matrix
-    arma::cx_mat U;
-    if(start==CANORB)
-      // Start with canonical orbitals
-      U.eye(orbidx.size(),orbidx.size());
-    else if(start==NATORB)
-      U=atomic_orbital_guess(basis,P,Cwrk);
-    else if(start==CHOLORB)
-      U=cholesky_guess(basis,Cwrk);
-    else if(start==ORTHMAT)
-      // Start with orthogonal matrix
-      U=std::complex<double>(1.0,0.0)*real_orthogonal(orbidx.size(),seed);
-    else if(start==UNITMAT)
-      // Start with unitary matrix
-      U=complex_unitary(orbidx.size(),seed);
-    else {
-      ERROR_INFO();
-      throw std::runtime_error("Starting point not implemented!\n");
+  arma::cx_mat Cplx(C*std::complex<double>(1.0,0.0)); // purpose?
```

Comment:
It's basically a cast from `arma::mat` to `arma::cx_mat`. Maybe `arma::cx_mat Cplx = arma::conv_to<arma::cx_mat>::from(C);` would also work.